internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,78 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	defaultBranch    = "main"
+)
+
+type CheckoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func NewCheckoutCmd(s v1alpha1.GitSource) CheckoutCmd {
+	return CheckoutCmd{GitSource: s}
+}
+
+func (c CheckoutCmd) String() (string, error) {
+	var checkoutCommand string
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
+	var commit = c.Ref.Commit
+	var tag = c.Ref.Tag
+
+	repositoryName, err := c.ValidateRepo()
+	if err != nil {
+		return "", err
+	}
+
+	if directory == "" {
+		directory = defaultDirectory
+	}
+
+	if branch == "" {
+		branch = defaultBranch
+	}
+
+	if commit != "" {
+		checkoutCommand = fmt.Sprintf("git clone %s && cd %s && git checkout %s && cp -r %s/* /manifests",
+			repository, repositoryName, commit, directory)
+		return checkoutCommand, nil
+	}
```

Comment:
It seems like this code would let me specify `branch`, `commit`, and `tag`, and then would silently ignore `branch` and `tag`.

Can we validate that exactly one of `branch` or `commit` or `tag` is set?

Reply:
IMO, we should do this as part of reconcile, and then also layer this same validation into the CRD schema or a webhook.

Reply:
Will add additional validation here, sure. 

Looking to add some union semantics to the git fields as well to help with validation at Bundle apply-time